haskell/test/Lorentz/Contracts/Test/Management.hs

```diff
@@ -0,0 +1,504 @@
+        validate . Right $ expectAnySuccess
+
+    it "fails if sender is not master minter" $ integrationalTestExpectation $ do
+      let
+        originationParams =
+            addMinter (wallet1, 0)
+          $ defaultOriginationParams
+      withOriginated originate originationParams $ \stablecoinContract -> do
+        withSender wallet2 $ lCallEP stablecoinContract (Call @"Remove_minter") wallet1
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "cannot remove the same wallet if it's already removed" $ integrationalTestExpectation $ do
+      let
+        originationParams =
+            addMinter (wallet1, 0)
+          $ defaultOriginationParams
+      withOriginated originate originationParams $ \stablecoinContract -> do
+        withSender admin $ lCallEP stablecoinContract (Call @"Remove_minter") wallet1
+        withSender admin $ lCallEP stablecoinContract (Call @"Remove_minter") wallet1
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "cannot remove non-minter" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        withSender admin $ lCallEP stablecoinContract (Call @"Remove_minter") wallet1
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+
+
+  describe "Minting" $ do
+    it "successfully mints tokens" $ integrationalTestExpectation $ do
+      let
+        originationParams =
+            addMinter (wallet1, 30)
+          $ defaultOriginationParams
+              { opPermissionsDescriptor = permissionDescriptorOperatorTransferAllowed }
+      withOriginated originate originationParams $ \stablecoinContract -> do
+        let
+          mintings =
+            [ (#to_ .! wallet1, #amount .! 10)
+            , (#to_ .! wallet2, #amount .! 5)
+            , (#to_ .! wallet3, #amount .! 15)
+            ]
+
+        withSender wallet1 $ lCallEP stablecoinContract (Call @"Mint") mintings
+
+        consumer <- lOriginateEmpty @[BalanceResponseItem] contractConsumer "consumer"
+        let
+          balanceRequestItems =
+            [ (#owner .! wallet1, #token_id .! 0)
+            , (#owner .! wallet2, #token_id .! 0)
+            , (#owner .! wallet3, #token_id .! 0)
+            ]
+          balanceRequest = mkView (#requests .! balanceRequestItems) consumer
+          balanceExpected =
+            [ (#request .! (#owner .! wallet1, #token_id .! 0), #balance .! 10)
+            , (#request .! (#owner .! wallet2, #token_id .! 0), #balance .! 5)
+            , (#request .! (#owner .! wallet3, #token_id .! 0), #balance .! 15)
+            ]
+
+        lCallEP stablecoinContract (Call @"Balance_of") balanceRequest
+
+        validate . Right $
+          lExpectViewConsumerStorage consumer [balanceExpected]
+
+    it "aborts whole transaction if one of the amount of minting tokens exceeds current minting allowance" $ integrationalTestExpectation $ do
```

Comment:
Why 'one of the amount', and not the sum of all?

Reply:
Changed